lib/puppet/module/task.rb

```diff
@@ -61,6 +62,33 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.find_files(name, metadata)
+      if metadata.key?('files')
+        file_list = metadata['files'].map do |file|
+          splits = file.split("/")
+          pup_module = Puppet::Module.find(splits[0])
```

Comment:
How does this work with environments? How does any of this work with environments?

Reply:
Looking at `Puppet::Module.find`, it appears to use `Puppet.lookup(:current_environment)`, which I think is safe in the context of a compile. So maybe that's how most of the rest of it works too.

Reply:
Yeah, this is what I'm working on getting working now!

Reply:
Change is a-coming ;)

Reply:
Ok, I think I addressed this.